### PR TITLE
docs: enable algolia search by adding structure

### PR DESCRIPTION
**Pull request by tara-det-ai**
Experiment Configuration Reference page: Add structure to enable algolia search index to find field names on the page.

### CODE REVIEW

File: docs/reference/reference-training/experiment-config-reference.rst [hunk 1]
```diff
@@ -1354,40 +1505,42 @@ The ``slurm`` section specifies configuration options applicable when the cluste
...
 ``slots_per_node``
-   The minimum number of slots required for a node to be scheduled during a trial. If
-   :ref:`gres_supported <cluster-configuration-slurm>` is false, specify ``slots_per_node`` in order
-   to utilize more than one GPU per node. It is the user’s responsibility to ensure that
-   ``slots_per_node`` GPUs will be available on the nodes selected for the job using other
-   configurations such as targeting a specific resource pool with only ``slots_per_node`` GPU nodes
-   or specifying a PBS constraint in the experiment configuration.
+==================
+
+Optional. ``slots_per_node`` specifies the minimum number of slots required for a node to be
```
Comment (rb-determined-ai) [inline docs/reference/reference-training/experiment-config-reference.rst:1541]: In other cases, you don't repeat the option name like you do in:

> Optional.  **slots_per_node** specifies...

File: docs/reference/reference-training/experiment-config-reference.rst [hunk 2]
```diff
@@ -1268,32 +1415,32 @@ of records).
 
 .. note::
 
-   If the amount of data to train a model on is specified using records or epochs and the batch size
-   does not divide evenly into the configured number of inputs, the remaining "partial batch" of
+   When the amount of training data for a model is specified using records or epochs, and the batch
```
Comment (rb-determined-ai) [inline docs/reference/reference-training/experiment-config-reference.rst:1418]: Note, the whole Training Units section explains a few different config fields, but has now config fields all to itself.

So it's weird that it's sandwiched between actual config fields in the hierarchy (Profiling and Slurm Options).

File: docs/reference/reference-training/experiment-config-reference.rst [hunk 3]
```diff
@@ -91,7 +100,7 @@ directory name is prefixed by ``<module>`` or used as the module if the ``<modul
 Arbitrary Script
 ================
 
-An arbitrary entrypoint script name.
+Required. An arbitrary entrypoint script name.
```
Comment (rb-determined-ai) [inline docs/reference/reference-training/experiment-config-reference.rst:103]: `entrypoint` is a single field, and it's required.  But "Abitrary Script", "Preconfigured Launch Module with Script", "Preconfigured Launch Module with Legacy Trial Definition", and "Legacy Trial Definition" are each different _ways_ of filling out that `entrypoint` required field.  So each different way shouldn't probably comment on if it's required or not, and they should likely be subsections of the `entrypoint` section.

**Review by rb-determined-ai — COMMENTED** [review]
I really like the removal of the `Optional` bits from the hierarchy.

I still don't think the hierarchy reflects the expconf schema yet.  Based on the right sidebar nav, here are the fields that reflect the hierarchy right now:

- [ ] Metadata
- [ ] Entrypoint
- [ ] Basic Behaviors
- [ ] Validation Policy
- [ ] Checkpoint Policy
- [ ] Checkpoint Storage
- [ ] Storage Type
- [x] Hyperparameters
- [x] Searcher
- [x] Resources
- [x] Bind Mounts
- [x] Environment
- [x] Optimizations
- [x] Reproducibility
- [x] Profiling
- [ ] Training Units
- [x] Slurm Options
- [x] PBS Options

Now, I am ok with approving the improvements made so far, but if you want to push for "doc hierarchy reflects schema" there's a lot of work left to do.  Since that's what you said you wanted previously, I'm going to not approve this yet.  But let me know if you want to land what you've got before doing a larger refactor.

File: docs/reference/reference-training/experiment-config-reference.rst (hunk 2 above)
Comment (tara-det-ai) [reply docs/reference/reference-training/experiment-config-reference.rst:1418]: agree. resolved by moving the Training Units section to the top.

File: docs/reference/reference-training/experiment-config-reference.rst (hunk 3 above)
Comment (tara-det-ai) [reply docs/reference/reference-training/experiment-config-reference.rst:103]: resolved by assigning level 4 headings (-------) to: "Arbitrary Script", "Preconfigured Launch Module with Script", "Preconfigured Launch Module with Legacy Trial Definition", and "Legacy Trial Definition"

**Review by rb-determined-ai — APPROVED** [review]
We're landing these changes as-is and doing another round of reorg to focus on the hierarchy.

**Comment by tara-det-ai** [comment]
[look at](https://github.com/determined-ai/determined/tree/main/master/pkg/schemas/expconf) /master/pkg/schemas/expconf/*.go, that might be a bit easier

**Pull request merged**